Difference with OpenBSD vi on handling filename completion

Hi!,
I'm using OpenVi on Mac.  It's great!
One big difference between OpenVi and OpenBSD's vi is when I use filec=<tab> to expand filenames.
On OpenVi I have to keep typing in an '\*' and then push tab if I want to expand what I have so far.
On OpenBSD vi it seems like the '\*' is assumed.  All I have to do i keep hitting tab as i type more specific
filenames, and it keeps matching.

Can you think of a way that I can not have to enter the '*' char to match all like in OpenBSD vi ?

This is great!
 
But when I do a checkout it doesn't compile:
 
 
 
***@***.***:~/code/c/OpenVi$ make
cc: ex/ex_argv.o
ex/ex_argv.c: In function 'argv_lexp':
ex/ex_argv.c:516:33: error: 'struct dirent' has no member named 'd_namlen'; did you mean 'd_name'?
516 | if (dp->d_namlen < nlen ||
| ^~~~~~~~
| d_name
ex/ex_argv.c:522:43: error: 'struct dirent' has no member named 'd_namlen'; did you mean 'd_name'?
522 | argv_alloc(sp, dlen + dp->d_namlen + 2);
| ^~~~~~~~
| d_name
ex/ex_argv.c:530:43: error: 'struct dirent' has no member named 'd_namlen'; did you mean 'd_name'?
530 | memcpy(p, dp->d_name, dp->d_namlen + 1);
| ^~~~~~~~
| d_name
ex/ex_argv.c:531:59: error: 'struct dirent' has no member named 'd_namlen'; did you mean 'd_name'?
531 | exp->args[exp->argsoff]->len = dlen + dp->d_namlen + 1;
| ^~~~~~~~
| d_name
make: *** [GNUmakefile:541: ex/ex_argv.o] Error 1

> On 05/20/2025 9:46 PM EDT Jeffrey H. Johnson ***@***.***> wrote:
>  
>  
> 
>  
> 
> Closed #22 https://github.com/johnsonjh/OpenVi/issues/22 as completed via 341738c https://github.com/johnsonjh/OpenVi/commit/341738c5baeae9bf3e3805b934ace8282144693e.
> 
> —
> Reply to this email directly, view it on GitHub https://github.com/johnsonjh/OpenVi/issues/22#event-17744076313, or unsubscribe https://github.com/notifications/unsubscribe-auth/AAAKG5VLW233YIWRZYJJEJ327PLHPAVCNFSM6AAAAABRL5TFDSVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXG42DIMBXGYZTCMY.
> You are receiving this because you authored the thread.Message ID: ***@***.***>
>

--- a/ex/ex_argv.c
+++ b/ex/ex_argv.c
@@ -502,7 +502,7 @@ argv_lexp(SCR *sp, EXCMD *excp, char *path)
                 }
                 name = p + 1;
         }
-        nlen = strlen(name) + 2;
+        nlen = strlen(name);
 
         if ((dirp = opendir(dname)) == NULL) {
                 msgq_str(sp, M_SYSERR, dname, "%s");
@@ -513,23 +513,22 @@ argv_lexp(SCR *sp, EXCMD *excp, char *path)
                         if (dp->d_name[0] == '.')
                                 continue;
                 } else {
-                        if (memcmp(dp->d_name, name, nlen))
+                        if (D_NAMLEN(dp) < nlen ||
+                            memcmp(dp->d_name, name, nlen))
                                 continue;
                 }
 
                 /* Directory + name + slash + NULL. */
-                argv_alloc(sp, dlen + nlen + 2);
+                argv_alloc(sp, dlen + D_NAMLEN(dp) + 2);
                 p = exp->args[exp->argsoff]->bp;
-                if (p == NULL)
-                        return (1);
                 if (dlen != 0) {
                         memcpy(p, dname, dlen);
                         p += dlen;
                         if (dlen > 1 || dname[0] != '/')
                                 *p++ = '/';
                 }
-                memcpy(p, dp->d_name, nlen + 1);
-                exp->args[exp->argsoff]->len = dlen + nlen + 1;
+                memcpy(p, dp->d_name, D_NAMLEN(dp) + 1);
+                exp->args[exp->argsoff]->len = dlen + D_NAMLEN(dp) + 1;
                 ++exp->argsoff;
                 excp->argv = exp->args;
                 excp->argc = exp->argsoff;

